test/distributed/org/apache/cassandra/distributed/impl/Instance.java

```diff
@@ -467,17 +465,23 @@ public void startup(ICluster cluster)
                 StorageService.instance.registerDaemon(CassandraDaemon.getInstanceForTesting());
                 if (config.has(GOSSIP))
                 {
+                    MigrationManager.setUptimeFn(() -> TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startedAt));
                     StorageService.instance.initServer();
                     StorageService.instance.removeShutdownHook();
                     Gossiper.waitToSettle();
                 }
                 else
                 {
-                    initializeRing(cluster);
+                    cluster.stream().forEach(peer -> {
+                        if (peer instanceof IInvokableInstance)
+                            GossipHelper.statusToNormal((IInvokableInstance) peer).accept(this);
+                        else
+                            GossipHelper.unsafeStatusToNormal((IInstance) peer).accept(this);
```

Comment:
we don't need to worry about this case, as cluster can only have `IInvokableInstance`; we can simplify this to

```
assert peer instanceof IInvokableInstance : "Peer is not a  IInvokableInstance";
GossipHelper.statusToNormal((IInvokableInstance) peer).accept(this);
```

Reply:
spoke in slack about this, the reason is UpgradeCluster

```
public interface IUpgradeableInstance extends IInstance {
    void setVersion(Version var1);
}
```

We use IInstance there so need the special case